filters/auth/jwt_validation_test.go

```diff
@@ -0,0 +1,177 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/net"
+	"github.com/zalando/skipper/proxy/proxytest"
+)
+
+const (
+	kid = "mykid"
+)
+
+var (
+	privateKey, _ = rsa.GenerateKey(rand.Reader, 2048)
+)
+
+func createToken(t *testing.T) string {
+	// Create the Claims
+	claims := &jwt.StandardClaims{
+		ExpiresAt: time.Now().Unix() + 1000,
+		Issuer:    "test",
+		Subject:   "aaa",
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+	token.Header["kid"] = kid
+
+	s, err := token.SignedString(privateKey)
+
+	t.Logf("Token: %v %v", s, err)
+
+	return s
+}
+
+func TestToken(t *testing.T) {
+	s := createToken(t)
+
+	parsedToken, err := jwt.Parse(s, func(token *jwt.Token) (interface{}, error) {
+		return &privateKey.PublicKey, nil
+	})
+
+	if err != nil {
+		t.Errorf("Failed to json decode: %v", err)
+		return
+	}
+
+	err = parsedToken.Claims.Valid()
+
+	if err != nil {
+		t.Errorf("Failed token: %v", err)
+		return
+	}
+
+}
+
+func TestJWTValidation(t *testing.T) {
+	cli := net.NewClient(net.Options{
+		IdleConnTimeout: 2 * time.Second,
+	})
+	defer cli.Close()
+
+	backend := httptest.NewServer(http.HandlerFunc(func(http.ResponseWriter, *http.Request) {}))
+	defer backend.Close()
+
+	authServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(489)
+			return
+		}
+
+		allKeys := map[string][]interface{}{}
+		allKeys["keys"] = append(allKeys["keys"], map[string]interface{}{"kid": kid, "n": base64.RawURLEncoding.EncodeToString(privateKey.PublicKey.N.Bytes()), "e": "AQAB"})
+		if r.URL.Path != testAuthPath {
+			w.WriteHeader(488)
+			return
+		}
+
+		e := json.NewEncoder(w)
+		err2 := e.Encode(allKeys)
+		if err2 != nil && err2 != io.EOF {
+			t.Errorf("Failed to json encode: %v", err2)
+		}
```

Comment:
```go
fmt.Fprintf(w, `{"keys":[{"kty":"RSA", "alg":"RS256", "kid": "%s", "n":"%s","e":"AQAB"}]}`, 
kid, base64.RawURLEncoding.EncodeToString(privateKey.PublicKey.N.Bytes()))
```
`AQAB` here is base64 of public exponent 65537 https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/crypto/rsa/rsa.go;l=260